test/Constraints/rdar38225184.swift

```diff
@@ -0,0 +1,4 @@
+// RUN: %target-typecheck-verify-swift
+
+extension Collection where Element == Int && Index == Int {}
+// expected-error {{use ',' for spliting conditions rather than '&&'}}
```

Comment:
This should diagnose on the line above, so it should be

`expected-error@-1`

Reply:
This test should also be in the Parse tests somewhere. Constraints is for the expression checker mostly.